src/filesystem.rs

```diff
@@ -44,13 +43,8 @@ pub fn is_existing_directory(path: &Path) -> bool {
     path.is_dir() && (path.file_name().is_some() || path.normalize().is_ok())
 }
 
-#[cfg(any(unix, target_os = "redox"))]
-pub fn is_executable(_: &Path, md: &fs::Metadata) -> bool {
-    md.permissions().mode() & 0o111 != 0
-}
-
-#[cfg(windows)]
-pub fn is_executable(path: &Path, _: &fs::Metadata) -> bool {
+/// Return true if the path point to a file executable by the current effective user
+pub fn is_executable(path: &Path) -> bool {
```

Comment:
This whole helper function is now completely unnecessary. I think we should simply use `.executable()` at the usage site?

Reply:
Good question, I'd be in favor of less lines is better, but it can also be more elegant and coherent to keep the check itself in the filesystem part.

Reply:
The `filesystem` module  is only here to add *additional* functionality that is not provided by the standard library (+ `faccess`, now). It was never intended to be a wrapper around all possible filesystem functionality. So I'd be in favor of removing `is_executable`.